airflow/models/dag.py

```diff
@@ -188,10 +188,10 @@ def get_last_dagrun(dag_id, session, include_externally_triggered=False):
     Overridden DagRuns are ignored.
     """
     DR = DagRun
-    query = select(DR).where(DR.dag_id == dag_id, DR.logical_date.is_not(None))
+    query = select(DR).where(DR.dag_id == dag_id, DR.run_after.is_not(None))
     if not include_externally_triggered:
         query = query.where(DR.external_trigger == expression.false())
-    query = query.order_by(DR.logical_date.desc())
+    query = query.order_by(DR.run_after.desc())
```

Comment:
We shouldn’t need to change this, filtering by logical date is correct here. We should review usages of this function if it causes problems, but “last” should be defined by logical date according to the AIP document.

Reply:
does that mean `the last dagrun` will never be an asset triggered dag?

Reply:
Which might be confusing as the "Recent Tasks" on UI will be empty if there's only such dag runs

Reply:
or should we include asset-triggered dag if nothing is found?

Reply:
After the latest revert and testing, seems to work fine 🤔  resolve this one

Reply:
https://github.com/apache/airflow/issues/46669 created